### PR TITLE
Websockets

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [ ] Maria
- [x] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [ ] Bug fix or refactor (change which fixes an issue)
- [x] New feature (change which adds functionality)

## Summary of the change.

Adds websockets setup for the chat and `shopping list` status change

Note any questions or struggles as well.

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -1 +1 @@
-web: gunicorn run:app
+web: gunicorn --worker-class eventlet -w 1 run:app
```
Comment (adumortier) [inline Procfile:1]: we'll see if this work, this command is listed in the docs when deploying a flask app with websockets handled through `eventlet` 

https://flask-socketio.readthedocs.io/en/latest/

File: pyproject.toml [hunk 2]
```diff
@@ -16,12 +16,13 @@ requests = "^2.23.0"
 ipdb = "^0.13.2"
 gunicorn = "^20.0.4"
 coverage = "^5.1"
+flask-socketio = "^4.3.0"
+eventlet = "^0.25.2"
```
Comment (adumortier) [inline pyproject.toml:20]: using the `flask-socketio` library and the `eventlet` service to implement the websockets

File: run.py [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+
+@socketio.on('connect')
+def on_connect():
+  print('just connected')
+
+@socketio.on('joinRoom')
+def on_join(data):
+    id = data['id']
+    print('id: ' + data['id'])
+    join_room(id)
```
Comment (adumortier) [inline run.py:22]: The code of the `on_join` method gets executed when a `joinRoom` event is sent by the front end

File: run.py [hunk 4]
```diff
@@ -1,10 +1,43 @@
...
+from flask_socketio import join_room, leave_room, emit
+from flask import request
+# from flask import Flask, render_template
 # do this once you're ready for production, etc:
 # config_name = os.getenv('APP_SETTINGS')
 config_name = "development"
 
 app = create_app(config_name)
+# app = Flask(__name__)
+socketio = SocketIO(app, cors_allowed_origins='*', logger=True, engineio_logger=True)
```
Comment (mronauli) [inline run.py:12]: '*' allows all origins to connect to the server

File: run.py [hunk 5]
```diff
@@ -1,10 +1,43 @@
...
+
+@socketio.on('leaveRoom')
+def on_leave(data):
+  print('about to leave room {}'.format(data['id']))
+  leave_room(id)
+
+@socketio.on('statusChange')
+def change_status(data):
+  room = data["id"]
+  print('changing status' + data["message"])
```
Comment (mronauli) [inline run.py:38]: i like that you display the status change for the user

**Pull request merged**